src/scripts/towns/DreamOrbController.ts

```diff
@@ -0,0 +1,119 @@
+class DreamOrbLoot {
+    constructor(public item: BagItem, public weight: number) {
+    }
+}
+
+class DreamOrb {
+    public amount = ko.observable(0);
+    constructor(public color: string, public requirement: Requirement, public items: DreamOrbLoot[]) {
+    }
+}
+
+class DreamOrbController implements Saveable {
+    public selectedOrb: KnockoutObservable<DreamOrb>;
+    public opening: KnockoutObservable<boolean>;
+    public item: KnockoutObservable<DreamOrbLoot>;
+
+    constructor() {
+        this.selectedOrb = ko.observable(this.orbs[0]);
+        this.opening = ko.observable(false);
+        this.item = ko.observable(undefined);
+    }
+
+    public orbs = [
+        new DreamOrb('Pink', undefined, [
+            new DreamOrbLoot({type: ItemType.item, id: 'Shuckle'}, 0.125),
+            new DreamOrbLoot({type: ItemType.item, id: 'Swablu'}, 0.125),
+            new DreamOrbLoot({type: ItemType.item, id: 'Riolu'}, 0.125),
+            new DreamOrbLoot({type: ItemType.item, id: 'Munna'}, 0.125),
+            new DreamOrbLoot({type: ItemType.item, id: 'Tornadus (Therian)'}, 0.075),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Sun Stone'}, 0.15),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Water Stone'}, 0.15),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Revive'}, 0.125),
+
+        ]),
+        new DreamOrb('Green', new ObtainedPokemonRequirement(pokemonMap['Tornadus (Therian)']), [
+            new DreamOrbLoot({type: ItemType.item, id: 'Drifloon'}, 0.15),
+            new DreamOrbLoot({type: ItemType.item, id: 'Bronzor'}, 0.15),
+            new DreamOrbLoot({type: ItemType.item, id: 'Sigilyph'}, 0.125),
+            new DreamOrbLoot({type: ItemType.item, id: 'Thundurus (Therian)'}, 0.075),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Red Shard'}, 0.125),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Blue Shard'}, 0.125),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Yellow Shard'}, 0.125),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Green Shard'}, 0.125),
+        ]),
+        new DreamOrb('Orange', new ObtainedPokemonRequirement(pokemonMap['Thundurus (Therian)']), [
+
+            new DreamOrbLoot({type: ItemType.item, id: 'Staryu'}, 0.133),
+            new DreamOrbLoot({type: ItemType.item, id: 'Ralts'}, 0.133),
+            new DreamOrbLoot({type: ItemType.item, id: 'Rotom'}, 0.134),
+            new DreamOrbLoot({type: ItemType.item, id: 'Landorus (Therian)'}, 0.075),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Fire Stone'}, 0.25),
+            new DreamOrbLoot({type: ItemType.item, id: 'Protein'}, 0.275),
+        ]),
+        //new DreamOrb('Blue', new ObtainedPokemonRequirement(pokemonMap['Enamorus (Therian)']), [new DreamOrbLoot({type: ItemType.item, id: 'Enamorus (Therian)'}, 0.01)]),
```

Comment:
I know that this is commented but, following the previous pattern, should the requirement not be Landorus instead of the Enamorus that we obtain in here ?
Also, Enamorus appears twice is this pool, making the sum of weights 1.01.

Reply:
Good catch, I'll give this a fix